viewing_party/party.py

```diff
@@ -0,0 +1,243 @@
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+#add movies to user_data watchlist and returns user_data
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+#modifies watched and watchlist lists if movie is watched
+#and returns user_data list
+def watch_movie(user_data, title):
+    #loop through each movie in watchlist
+    #if movie title is same as given title, add movie to watched
+    #remove movie from watchlist
+    for movie in user_data['watchlist']:
+        if movie['title'] == title:
+            user_data['watched'].append(movie)
+            user_data['watchlist'].remove(movie)
+    return user_data
+
+######################################################
+                # Wave 2 #
+######################################################
+
+#calculates average for all movie ratings and returns average rating
+def get_watched_avg_rating(user_data):
+    #initliaze list
+    rating_list = []
+    
+    #check if watched list has at least one movie
+    #loop through each movie in watched list and get rating
+    #calculates avg rating or return 0 if no movies are in list
+    if len(user_data['watched']) >= 1:
+        for movies in user_data['watched']:
+            rating_list.append(movies['rating'])
+        avg_rating = sum(rating_list)/len(rating_list)
+    else:
+        avg_rating = 0
+    
+    return avg_rating
+
+
+#returns the most watched genre
+def get_most_watched_genre(user_data):
+    #call make_genre_frequency_dict and store frequency dict in genre_frequency
+    genre_frequency = make_genre_frequency_dict(user_data)
+    if len(genre_frequency) > 0:
+        popular_genre = max(genre_frequency, key = genre_frequency.get)
+        return popular_genre
+    else:
+        return None
+
+#takes user data to create genre frequency and returns it
+def make_genre_frequency_dict(user_data):
+    #intialize variables
+    genre_frequency = {}
+
+    #loop through movies in watched list and checks if movie genre is key
+    #add 1 to value in dict if genre is key
+    #add genre to freq_dict if not in freq_dict
+    for movies in user_data['watched']:
+        if movies['genre'] in genre_frequency.keys():
+            genre_frequency[movies['genre']] += 1
+        else:
+            genre_frequency[movies['genre']] = 1
+    
+    return genre_frequency
+
+######################################################
+                # Wave 3 #
+######################################################
+
+#returns list of dictionary that user has seen that none of their friends have seen
+def get_unique_watched(user_data):
+    #initalize variables
+    user_unqiue_movie_watched = []
+
+    #call set-making functions and capture return in assigned var
+    my_movie_set = make_my_movie_set(user_data) 
+    friend_movie_set = make_friend_movie_set(user_data)
```

Comment:
Great use of helper functions to DRY (don't repeat yourself) up your code and make it clear